tools/awslint/lib/rules/construct.ts

```diff
@@ -227,4 +227,29 @@ constructLinter.add({
       e.assert(!property.name.toLowerCase().endsWith('arn'), `${e.ctx.propsFqn}.${property.name}`);
     }
   }
+});
+
+constructLinter.add({
+  code: 'props-no-tokens',
+  message: 'props should not use the "Token" type',
+  eval: e => {
+    if (!e.ctx.propsType) { return; }
+    if (!e.ctx.hasPropsArgument) { return; }
+
+    // this rule only applies to L2 constructs
+    if (e.ctx.classType.name.startsWith('Cfn')) { return; }
+
+    for (const property of e.ctx.propsType.ownProperties) {
```

Comment:
why `ownProperties` and not `allProperties`?

Reply:
good catch - updated.